doc/02_development/11_dvc.md

```diff
@@ -0,0 +1,167 @@
+    * [Storage](#storage)
+  * [Using DVC](#using-dvc)
+    * [Add file or folder to DVC](#add-file-or-folder-to-dvc)
+    * [Pull / push](#pull--push)
+      * [Authentication](#authentication)
+    * [Python API](#python-api)
+  * [Example](#example)
+<!-- TOC -->
+
+## General
+
+DVC enables us to manage our datasets and trained models outside from git.
+Instead of storing the full datasets or models inside git, DVC will just store a reference to the dataset or trained model stored in a separate storage.
+This makes it possible to switch between versions of datasets and trained models and train models without touching the git repo.
+
+![DVC flow](https://dvc.org/img/flow.gif)
+
+(Taken from <https://dvc.org/>)
+
+A more detailed summary about what it does cant be found [here (german)](https://www.marianbiermann.de/data-version-control-reproduzierbares-machine-learning/).
+Additional information can also be found in the according [GitHub repsoitory](https://github.com/iterative/dvc)
+
+### Installation
+
+Install DVC and the according gdrive extension by running `pip install dvc dvc-gdrive`.
+
+### Storage
+
+Currently, we use Google Drive as repo for DVC.
+Pull access is available for everyone, since the storage folder is accessible for everyone with the link.
+
+However, if you want to push changes you need write access.
+An administrator has to add your Google Account by doing the following.
+
+1. Go to `https://drive.google.com` and login with our user
+2. Click the folder `paf22`:
+
+![paf22 folder](../00_assets/gdrive-paf.png)
+3. click on `Manage permissions` on the right side
+4. Add the user as `Collaborator`
+
+![paf22 folder](../00_assets/gdrive-permissions.png)
+
+## Using DVC
+
+### Add file or folder to DVC
+
+To add a file or folder to DVC use the following commands:
+
+```shell
+# Add file data/data.xml
+dvc add data/data.xml
+
+# Add the whole data directory
+dvc add data
+```
+
+   > ❗️ if you already added the directory to git you have to remove it by running `git rm -r --cached '<file or foldername>'`
+
+### Pull / push
+
+The commands to pull and push are similar to git. However, for pushing an authentication may be required.
+
+```shell
+# pull
+dvc pull
+
+# push
+dvc push
+
+# checkout (run after doing git checkout)
+dvc checkout
+```
+
+#### Authentication
+
+When pushing the first time you will receive a message similar to this:
+
+```shell
+Your browser has been opened to visit:
+
+https://accounts.google.com/o/oauth2/auth?client_id=xxxxxxxxxxxx
+```
+
+To authenticate please click the link and login with your Google account.
+
+> ❗️ Your Google account has to be added to the users with write access. Please contact [@timdreier](https://www.github.com/timdreier) or [@ll7](https://github.com/ll7).
+
+After this, you should get a success message similar to this in your browser:
+
+```text
+The authentication flow has completed.
+```
+
+Back in your shell, you should see that the push was successful.
+
+### Python API
```

Comment:
Maybe it would be helpful to add some more examples for the Python API like loading/saving models or checkpoints (model + epochs + optimizer)? 😊
Would save time, not having to search through the documentation on the website for these informations. Plus the documentation there is not really good in my opinion 😅
Or what do you think?

Reply:
Sure. However, since setting up DVC already consumed a lot of time, I didn't had time to do research about this.
In my opinion it would be best to do this in a separate PR in the next sprint.

That way, this PR can get merged so we can use dvc already for the datasets.

Reply:
Yes, you're right :)